client/components/Settings.js

```diff
@@ -15,20 +15,48 @@ import CustomBankField from './CustomBankField';
 import {OpCatChartTypeSelect, OpCatChartPeriodSelect} from './Charts';
 import T from './Translated';
 
-
 class Account extends React.Component {
 
+    constructor(props) {
+        super(props);
+        this.state = {
+            isDefault: (store.getDefaultAccountId() === this.props.account.id)
```

Comment:
Remove the state and use this.forceUpdate() to re-render in store.on(State.defaultAccount, func)